src/earn/poolInfoScreen/BeforeDepositBottomSheet.test.tsx

```diff
@@ -17,139 +19,99 @@ const mockToken = {
-    const { getByTestId } = render(
-      <Provider store={createMockStore()}>
-        <BeforeDepositBottomSheet
-          forwardedRef={{ current: null }}
-          token={mockToken}
-          pool={mockEarnPositions[0]}
-          hasDepositToken={true}
-          hasTokensOnSameNetwork={true}
-          hasTokensOnOtherNetworks={true}
-          canAdd={true}
-          exchanges={[]}
-        />
-      </Provider>
-    )
-    expect(getByTestId('Earn/ActionCard/Deposit')).toBeTruthy()
-    expect(getByTestId('Earn/ActionCard/SwapAndDeposit')).toBeTruthy()
-    expect(getByTestId('Earn/ActionCard/CrossChainSwap')).toBeTruthy()
-  })
-  it('show bottom sheet correctly when hasDepositToken is false, can same and cross chain swap', () => {
-    const { getByTestId } = render(
-      <Provider store={createMockStore()}>
-        <BeforeDepositBottomSheet
-          forwardedRef={{ current: null }}
-          token={mockToken}
-          pool={mockEarnPositions[0]}
-          hasDepositToken={false}
-          hasTokensOnSameNetwork={true}
-          hasTokensOnOtherNetworks={true}
-          canAdd={true}
-          exchanges={[]}
-        />
-      </Provider>
-    )
-    expect(getByTestId('Earn/ActionCard/SwapAndDeposit')).toBeTruthy()
-    expect(getByTestId('Earn/ActionCard/CrossChainSwap')).toBeTruthy()
-    expect(getByTestId('Earn/ActionCard/Add')).toBeTruthy()
-  })
+jest.mock('src/statsig')
 
-  it('show bottom sheet correctly when hasDepositToken is false, can cross chain swap', () => {
-    const { getByTestId } = render(
-      <Provider store={createMockStore()}>
-        <BeforeDepositBottomSheet
-          forwardedRef={{ current: null }}
-          token={mockToken}
-          pool={mockEarnPositions[0]}
-          hasDepositToken={false}
-          hasTokensOnSameNetwork={false}
-          hasTokensOnOtherNetworks={true}
-          canAdd={true}
-          exchanges={[]}
-        />
-      </Provider>
-    )
-    expect(getByTestId('Earn/ActionCard/Swap')).toBeTruthy()
-    expect(getByTestId('Earn/ActionCard/Add')).toBeTruthy()
+describe('BeforeDepositBottomSheet', () => {
+  beforeEach(() => {
+    jest.clearAllMocks()
+    jest
+      .mocked(getFeatureGate)
+      .mockImplementation((gate) => gate === StatsigFeatureGates.ALLOW_CROSS_CHAIN_SWAPS)
   })
 
-  it('show bottom sheet correctly when hasDepositToken is false, no tokens', () => {
-    const { getByTestId } = render(
-      <Provider store={createMockStore()}>
-        <BeforeDepositBottomSheet
-          forwardedRef={{ current: null }}
-          token={mockToken}
-          pool={mockEarnPositions[0]}
-          hasDepositToken={false}
-          hasTokensOnSameNetwork={false}
-          hasTokensOnOtherNetworks={false}
-          canAdd={true}
-          exchanges={[]}
-        />
-      </Provider>
-    )
-    expect(getByTestId('Earn/ActionCard/Add')).toBeTruthy()
-    expect(getByTestId('Earn/ActionCard/Transfer')).toBeTruthy()
-  })
+  it.each`
```

Comment:
These are a LOT, but much more thorough. This should get simpler with cross chain swap and deposit as the number of variants reduce significantly as we don't need to differentiate between having tokens on same chain or different chain.